### DISCUSSION TITLE
Make it possible to run kured on ARM nodes or allow disabling kured (to install it externally from helm)

### DISCUSSION THREAD

**Discussion by janosmiko** [mysticaltech/terraform-hcloud-kube-hetzner#1190] [Q&A]
### Description

Hi,

I just noticed if I add a taint to a nodepool (because I run arm instances in it), kured will not run on them.

```
  agent_nodepools = [
    {
      name        = var.mgmt_agent_server_name,
      server_type = "cax31",
      location    = "fsn1",
      labels      = [],
      taints      = [],
      count       = 1,
      labels      = [
        "kubernetes.io/arch=arm64"
      ],
      taints = [
        "kubernetes.io/arch=arm64:NoSchedule"
      ],
    },
  ]
```

To solve this issue, I can imagine two solutions.
- One is to make it possible to pass tolerations to the kured daemonset or install it from the kured helm chart and make it possible to pass values.
- The other is to make it possible to disable kured and allow installing it externally.

**Comment by mysticaltech** [comment]
@janosmiko It's not about ARM, it already runs great on such nodes, it's about the noschedule taint. You have a variable to add custom kured config, see kube.tf, and https://kured.dev/docs/configuration/ for options. Not sure if it can resolve this, but you should search the kured repo or have a chat with gpt-4 about tolerations. You use case is independent of our project. I will move this to a discussion.

**Comment by janosmiko** [comment]
@mysticaltech this answer is useless. 

I wrote exactly the same in the description of the feature request.
You cannot specify the tolerations of the kured pods using the kured options.

The daemonset is hardcoded. And that's the reason I asked for this feature.
https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/f2de7ad01aafc733414ed0100e939a13d8889957/templates/kured.yaml.tpl#L12

I really don't understand why you handle your users this harsh.

You even said that you are not sure if your suggestion can solve the issue. It cannot. The solution would be to add an option to configure the tolerations of the daemonset, or to disable the installation of kured by this terraform module. Because even if I'd install it externally, if I re-apply terraform, that's going to override the custom installation.

> You use case is independent of our project.

This is not true.

**Comment by mysticaltech** [comment]
@janosmiko I'm going to assume you were having a bad day yesterday.

Thanks for the details and clarity on the daemonset. That makes sense.

Please don't hesitate to submit a PR with a fix for your use case.

I remain open to helping you if you have more questions. Note that this is a open-source project, we do our best to help each other.

**Comment by mysticaltech** [comment]
FYI, we load kured via a manifest using a kustomization. Search kured in the project, you will understand.

It's possible that you can modify this manifest via tweaking the kustomization. Try it.

Try it, if you can make it work, I can guide you further to turn this to a PR.

See the debug and contribution section in the readme, it should help.
